src/components/clusters/NewClusterPage.tsx

```diff
@@ -40,14 +40,24 @@ type NewClusterFormProps = {
 
 const NewClusterForm: React.FC<NewClusterFormProps> = ({ pullSecret = '' }) => {
   const { addAlert, clearAlerts } = React.useContext(AlertsContext);
-  const { versions, getDefaultVersion } = React.useContext(OpenshiftVersionsContext);
+  const {
+    error: errorOCPVersions,
+    loading: loadingOCPVersions,
+    versions,
+    getDefaultVersion,
+  } = useOpenshiftVersions();
```

Comment:
I'd move this into `NewCluster` component (same as we have pullSecret fetching) and pass the `versions` to a `NewClusterForm` as props. Doing this allows the versions prop to be always defined (`[]` in case of an error) and avoids us having to use `enableReinitialize` on the form.

Reply:
Done. New `OpenshiftVersionSelect` component is introduced and all this logic co-located there